src/modules/peek/Peek.UI/MainWindowViewModel.cs

```diff
@@ -63,50 +126,124 @@ public void Initialize(HWND foregroundWindowHandle)
+
+        private void Navigate(NavigationDirection direction, bool isAfterDelete = false)
         {
             if (NavigationThrottleTimer.IsEnabled)
             {
                 return;
             }
 
-            NavigationThrottleTimer.Start();
+            if (Items == null || Items.Count == _deletedItemIndexes.Count)
+            {
+                _currentIndex = DisplayIndex = 0;
+                CurrentItem = null;
+                return;
+            }
+
+            _navigationDirection = direction;
+
+            int offset = direction == NavigationDirection.Forwards ? 1 : -1;
+
+            do
+            {
+                _currentIndex = MathHelper.Modulo(_currentIndex + offset, Items.Count);
+            }
+            while (_deletedItemIndexes.Contains(_currentIndex));
+
+            CurrentItem = Items[_currentIndex];
+
+            // If we're navigating forwards after a delete operation, the displayed index does not
+            // change, e.g. "(2/3)" becomes "(2/2)".
+            if (isAfterDelete && direction == NavigationDirection.Forwards)
+            {
+                offset = 0;
+            }
+
+            DisplayIndex = MathHelper.Modulo(DisplayIndex + offset, DisplayItemCount);
 
-            var itemCount = Items?.Count ?? 1;
-            CurrentIndex = MathHelper.Modulo(CurrentIndex - 1, itemCount);
-            CurrentItem = Items?.ElementAtOrDefault(CurrentIndex);
+            NavigationThrottleTimer.Start();
         }
 
-        public void AttemptNextNavigation()
+        /// <summary>
+        /// Sends the current item to the Recycle Bin.
+        /// </summary>
+        public void DeleteItem()
         {
-            if (NavigationThrottleTimer.IsEnabled)
+            if (CurrentItem == null || !IsFilePath(CurrentItem.Path))
             {
                 return;
             }
 
-            NavigationThrottleTimer.Start();
+            _deletedItemIndexes.Add(_currentIndex);
+            OnPropertyChanged(nameof(DisplayItemCount));
+
+            string path = CurrentItem.Path;
+
+            DispatcherQueue.GetForCurrentThread().TryEnqueue(() =>
+            {
+                Task.Delay(DeleteDelayMs);
+                DeleteFile(path);
+            });
 
-            var itemCount = Items?.Count ?? 1;
-            CurrentIndex = MathHelper.Modulo(CurrentIndex + 1, itemCount);
-            CurrentItem = Items?.ElementAtOrDefault(CurrentIndex);
+            Navigate(_navigationDirection, isAfterDelete: true);
+        }
+
+        private void DeleteFile(string path, bool permanent = false)
+        {
+            SHFILEOPSTRUCT fileOp = new()
+            {
+                wFunc = FO_DELETE,
+                pFrom = path + "\0\0",
+                fFlags = (ushort)(FOF_NOCONFIRMATION | (permanent ? 0 : FOF_ALLOWUNDO)),
+            };
+
+            int result = SHFileOperation(ref fileOp);
+
+            if (result != 0)
```

Comment:
We need a message (translated) for the user in peek ui. (Imagin youndelete files, nothing hapoens and you don't know why.) We can not expect that users checking thenlog file.

Suggestion: Infobar or teaching tip with severity error.

Reply:
I was in two minds about this while developing. It was a deliberate decision in the end to have the delete operation as a 'best effort' attempt which occurs in the background after navigating away from the current file. If the delete doesn't succeed, I didn't want to pop up a message saying anything technical.

However, I agree that there's the potential for this to be confusing, as the file will have been removed from the preview sequence, but still be on disk.

As you say, maybe the solution is to show a non-modal message.

If the delete fails, do you think the file should be reinstated in the preview sequence so it can be previewed again?

Reply:
Not readding it is confusing as an existing file isn't previewable anymore until reopening peek.